flask_admin/contrib/fileadmin/azure.py

```diff
@@ -160,80 +152,78 @@ def get_breadcrumbs(self, path):
 
         accumulator = []
         breadcrumbs = []
-        for folder in path.split(self.separator):
-            accumulator.append(folder)
-            breadcrumbs.append((folder, self.separator.join(accumulator)))
+        if path is not None:
+            for folder in path.split(self.separator):
+                accumulator.append(folder)
+                breadcrumbs.append((folder, self.separator.join(accumulator)))
         return breadcrumbs
 
     def send_file(self, file_path):
-        file_path = self._ensure_blob_path(file_path)
-
-        if not self._client.exists(self._container_name, file_path):
-            raise ValueError()
-
-        now = datetime.utcnow()
-        url = self._client.make_blob_url(self._container_name, file_path)
-        sas = self._client.generate_blob_shared_access_signature(
-            self._container_name,
-            file_path,
-            BlobPermissions.READ,
-            expiry=now + self._send_file_validity,
-            start=now - self._send_file_lookback,
+        path = self._ensure_blob_path(file_path)
+        if path is None:
+            raise ValueError("No path provided")
+        blob = self._container_client.get_blob_client(path).download_blob()
+        if not blob.properties or not blob.properties.has_key("content_settings"):
+            raise ValueError("Blob has no properties")
+        mime_type = blob.properties["content_settings"]["content_type"]
+        blob_file = io.BytesIO()
```

Comment:
I think the `blob_file` should be closed here. It will delete the buffer. `flask.send_file` doesn't seem to close it. It will be closed when the variable is GC'ed but if you're getting lots of files it could consume a lot of memory

Reply:
FYI, according to comments from David Lord of Flask issues, the file does eventually get closed: https://github.com/pallets/flask/issues/2468
I also did try the context manager approach, but immediately got an error from werkzeug being unable to access a closed file. So I've left this as is.